src/sentry/discover/endpoints/serializers.py

```diff
@@ -177,7 +177,7 @@ def validate_projects(self, projects):
             return projects
 
         # Params will have project_ids that have been validated
-        if projects != set(self.context["params"]["project_id"]):
+        if not projects.intersection(set(self.context["params"]["project_id"])):
```

Comment:
Will this still fail if projects from outside the organization are requested to be added to the query?

Reply:
Yep `test_save_with_wrong_projects` checks this, in the case the query is `project:some_project_from_another_org`

`projects` here would still have the other org's project but
`self.context["params"]["project_id"]` will only have projects for the user's org

Reply:
Yes the half valid, half invalid scenario is the one I was thinking of.

Reply:
🙏 Thanks, updated to avoid allowing that case